### PR TITLE
feat: added unit tests for deposit and withdrawal functionalities.

### LINKED ISSUE #5: feat: unit tests for Deposit and withdrawal functionalities.
Write a complete and an extensive set of unit-tests for the strapex contracts.
Scope of the tests: Deposit and withdrawal functionalities.

> [!NOTE]  
> As this contract will hold users funds it is highly important that  the test cover all possible edge cases and scenarios.

Some of examples of cairo unit tests implemented in other repos: 
- [ZkRamp](https://github.com/keep-starknet-strange/zkramp/blob/main/contracts/src/contracts/ramps/revolut/revolut_test.cairo)

Some interesting resources to learn about cairo testing:
 -[ Testing Cairo Programs](https://www.youtube.com/watch?v=FwtvBjJi4Bg&t=1949s&ab_channel=HorusLabs)

# Instructions to apply
1. Introduce yourself, tell about your background, etc.
2. Describe shortly how you plan to tackle the issue

### PR DESCRIPTION
**Pull request by SoarinSkySagar**
resolves #5

### CODE REVIEW

File: contracts/tests/test_strapex_contract.cairo [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        contract_address_const::<'OWNER'>().into(),
+        child_contract_felt,
+        contract_address_const::<'DEPOSIT_TOKEN'>().into()
+    ];
+    let contract_address = factory_contract.deploy(@calldata).expect('Deployment failed');
+
+    contract_address
+}
+
+#[test]
```
Comment (SoarinSkySagar) [inline contracts/tests/test_strapex_contract.cairo:31]: I'm encountering this issue here, @machuwey can you PTAL:
image

Comment (machuwey) [reply contracts/tests/test_strapex_contract.cairo:31]: We have to add 
```
[dev-dependencies]
cairo_test = "2.8.4"
```
as a dependency.

Consider also updating to lastest sn_forge version.

**Comment by SoarinSkySagar** [comment]
> Hello @SoarinSkySagar, could please rollback al changes made to `src/` contract files. Just to keep this PR clean an tidy and solely focused on the `tests/`. 
> 
> Thank you.

yeah sure @machuwey, however most of the changes are from the `scarb fmt` command which lints all cairo code to increase code readability and neatness (does not change any code), do you want me to rollback that too?

**Comment by machuwey** [comment]
> > Hello @SoarinSkySagar, could please rollback al changes made to `src/` contract files. Just to keep this PR clean an tidy and solely focused on the `tests/`.
> > Thank you.
> 
> yeah sure @machuwey, however most of the changes are from the `scarb fmt` command which lints all cairo code to increase code readability and neatness (does not change any code), do you want me to rollback that too?

Yes please, I solved all the warnings in a previous PR, if you want make another PR for the code linting that's ok for me. Thanks

**Comment by SoarinSkySagar** [comment]
> 
> Yes please, I solved all the warnings in a previous PR, if you want make another PR for the code linting that's ok for me. Thanks

@machuwey okay, creating a new PR only for linting

**Comment by SoarinSkySagar** [comment]
@machuwey done, all unnecessary changes rolled back

**Review by machuwey — APPROVED** [review]
Looks good to me, nice job @SoarinSkySagar in laying out the foundation to test the contracts.
